url47087446e9_kellrott_SparqlScape_tgz-pJ8-org_json_TestJ8/src/org/sparqlscape/SparqlScapePlugin.java

```diff
@@ -0,0 +1,241 @@
+			
+			JPanel fetchPanel = new JPanel();
+			Button fetchSelectedButton = new Button("Fetch Selected");
+			fetchSelectedButton.addActionListener( this );
+			fetchPanel.add(fetchSelectedButton);
+			filterCheckbox = new Checkbox("Filter", true);
+			fetchPanel.add(filterCheckbox);
+			
+			c.gridx = 0;
+			c.gridy = 4;
+			this.add( fetchPanel, c  );
+			
+			c.gridy = 5;
+			Button editNamespaces = new Button("Edit Namespaces");
+			editNamespaces.addActionListener(this);
+			this.add( editNamespaces, c );
+						
+			this.setVisible(true);			
+			
+			iSparql = new SparqlInterface( prefs.getEndpoint() );
+		}
+
+		@SuppressWarnings("unchecked")
+		@Override
+		public void actionPerformed(ActionEvent ae) {
+			String action = ae.getActionCommand();	
+						
+			LinkTriple []addNodes = null;
+			if ( action.equals("Fetch") ) {
+				addNodes = iSparql.ExploreURI( urlText.getText() );
+				addLinks(addNodes);			
+				//Cytoscape.getCurrentNetwork().addNode( Cytoscape.getCyNode( urlText.getText(), true ) );
+			}
+			if ( action.equals("Fetch Selected") ) {
+				CyNetwork network = Cytoscape.getCurrentNetwork();
+				Set nodeset = network.getSelectedNodes();
+				java.util.List<LinkTriple> nodeList = new LinkedList<LinkTriple>();
+				for (Object o : nodeset ) {
+					CyNode node = (CyNode) o;
+					for ( LinkTriple newNode : iSparql.ExploreURI(node.getIdentifier()) ) {
+						nodeList.add(newNode);
+					}
+				}
+				if ( filterCheckbox.getState() )
+					addLinks( iSparql.NodeFilter(null, nodeList.toArray(new LinkTriple[0]) ) );
+				else
+					addLinks( nodeList.toArray(new LinkTriple[0]) );
+					
+			}			
+			if ( action.equals("Find Nodes") ) {
+				CyAttributes nodeAttr = Cytoscape.getNodeAttributes();
+
+				CyNetwork network = Cytoscape.getCurrentNetwork();
+				Set nodeset = network.getSelectedNodes();
+				java.util.List<Map<String,Object>> nodeList = new LinkedList<Map<String,Object>>();
+				for (Object o : nodeset ) {
+					CyNode node = (CyNode) o;
+					Map<String, Object> map = new HashMap<String,Object>();
+					String id = node.getIdentifier();
+					map.put( "ID", id );
+					for ( String attr : nodeAttr.getAttributeNames() ) {
+						if ( nodeAttr.hasAttribute(id, attr )) {
+							System.out.println(id + " " + attr + " " + nodeAttr.getType(attr) );
+							if ( nodeAttr.getType(attr) == CyAttributes.TYPE_STRING ) {
+								Object iAttr = nodeAttr.getAttribute( id, attr );
+								map.put( attr, iAttr );
+							}
+						}
+					}
+					nodeList.add( map);
+				}				
+				LinkTriple [] links = iSparql.NodeSearch(null, nodeList, "ID" );
+				addLinks(links);
+			}
+			if ( action.equals("Edit Namespaces") ) {
+				Map<String,String> nameMap = prefs.getNameSpaces();
+				Map<String,String> newMap = NameSpaceEditor.EditDialog(nameMap);				
+				for ( String key : nameMap.keySet() ) {
+					prefs.removeNameSpace(key);
+				}
+				for ( String key : newMap.keySet() ) {
+					prefs.addNameSpace(key, newMap.get(key) );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.